File: config/settings/tinymce.py
''' 
Настройки ПО TinyMCE. TinyMCE - это WYSIWYG текстовый редактор. 
В данном Django проекте он используется при создании записей в блог. (через административную панель)
В данном конфиге определены подключаемые плагины TinyMCE и меню редактора.
'''
TINYMCE_DEFAULT_CONFIG = {
    'cleanup_on_startup': True,
    'custom_undo_redo_levels': 20,
    'selector': 'textarea',
    'branding': False,
    'plugins': '''
            textcolor save link image media preview codesample contextmenu
            table code lists fullscreen  insertdatetime  nonbreaking
            contextmenu directionality searchreplace wordcount visualblocks
            visualchars code fullscreen autolink lists  charmap print  hr
            anchor pagebreak autoresize
            ''',
    'toolbar1': '''
            fullscreen preview bold italic underline | fontselect,
            fontsizeselect  | forecolor backcolor | alignleft alignright |
            aligncenter alignjustify | indent outdent | bullist numlist table |
            | link image media | codesample |
            ''',
    'toolbar2': '''
            visualblocks visualchars |
            charmap hr pagebreak nonbreaking anchor | code |
            ''',
    'contextmenu': 'formats | link image',
    'menubar': True,
    'statusbar': True,
}

File: frontend/src/utils/tinymce.js
/**
 * Общая конфигурация WYSIWYG редактора TinyMCE для форм сайта.
 *
 * Ресурсы TinyMCE загружаются локально (tinymceScriptSrc /static/tinymce/),
 * без CDN и API-ключа; скины и стили содержимого входят в комплект пакета
 * и копируются сборкой в /static/tinymce/skins/.
 */

/**
 * Возвращает объект init для компонента Editor (@tinymce/tinymce-react)
 * в цветовой схеме, соответствующей теме сайта.
 *
 * @param {boolean} isDark - Действует ли темная тема сайта
 * @return {Object} Конфигурация init для TinyMCE
 *
 * @example
 * <Editor init={getEditorInit(isDark)} />
 */
export const getEditorInit = (isDark) => ({
  license_key: 'gpl',
  // Тема редактора следует теме сайта: скин интерфейса и стили содержимого
  menubar: true,
  statusbar: true,
  branding: false,
  promotion: false,
  skin: isDark ? 'oxide-dark' : 'oxide',
  content_css: isDark ? 'dark' : 'default',
  // Контент рендерится на страницах другой глубины, чем страница редактора:
  // относительные адреса разрешались бы в другие места. Адреса сохраняются
  // от корня сайта, без хоста
  relative_urls: false,
  remove_script_host: true,
  plugins: [
    'link', 'image', 'media', 'preview', 'codesample',
    'table', 'code', 'lists', 'fullscreen', 'insertdatetime', 'nonbreaking',
    'directionality', 'searchreplace', 'wordcount', 'visualblocks',
    'visualchars', 'autolink', 'charmap', 'anchor', 'pagebreak', 'autoresize',
  ],
  toolbar1: 'fullscreen preview bold italic underline | fontfamily fontsize | forecolor backcolor | alignleft alignright | aligncenter alignjustify | indent outdent | bullist numlist table | link image media | codesample',
  toolbar2: 'visualblocks visualchars | charmap hr pagebreak nonbreaking anchor | code',
});

export default getEditorInit;
